File: frontend/cook-book-app/src/model/recipeDetail.ts
import { CookingIngredient } from "./cookingIngredient";
import { CookingStep } from "./cookingStep";

export type RecipeDetail = {
    Id: string;
    RecipeId: string;
    Steps: CookingStep[];
    Ingredients: CookingIngredient[];
    TimeToCook: number;
    Servings: number;

}

File: backend/Cookbook/Cookbook.Domain/Model/RecipeDetail.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Cookbook.Domain.Model
{
    public class RecipeDetail
    {
        public Guid Id { get; set; }

        public List<CookingStep> Steps { get; set; }

        public List<Ingredient> Ingredients { get; set; }

        public List<RecipeNotes>? Notes { get; set; }

        public int? NumberOfPortions { get; set; }

    }
}